app/src/main/java/com/ncnf/user/User.java

```diff
@@ -1,203 +1,240 @@
 package com.ncnf.user;
 
 import com.google.firebase.auth.FirebaseAuth;
-import com.google.firebase.firestore.DocumentSnapshot;
-import com.ncnf.Utils;
-import com.ncnf.database.DatabaseResponse;
 import com.ncnf.database.DatabaseService;
 import com.ncnf.event.Event;
-import com.ncnf.event.EventBuilder;
 import com.ncnf.event.PrivateEvent;
 import com.ncnf.event.PublicEvent;
+import com.ncnf.utilities.InputValidator;
 
 import java.util.ArrayList;
-import java.util.HashMap;
+import java.util.Collections;
+import java.util.Date;
 import java.util.List;
-import java.util.Map;
 import java.util.Objects;
 import java.util.concurrent.CompletableFuture;
-import java.util.stream.Collectors;
 
-import static com.ncnf.Utils.BIRTH_YEAR_KEY;
-import static com.ncnf.Utils.EMAIL_KEY;
-import static com.ncnf.Utils.EMPTY_STRING;
+import static com.ncnf.Utils.EVENTS_COLLECTION_KEY;
 import static com.ncnf.Utils.FIRST_NAME_KEY;
-import static com.ncnf.Utils.FRIENDS_KEY;
-import static com.ncnf.Utils.LAST_NAME_KEY;
 import static com.ncnf.Utils.NOTIFICATIONS_KEY;
 import static com.ncnf.Utils.NOTIFICATIONS_TOKEN_KEY;
 import static com.ncnf.Utils.OWNED_EVENTS_KEY;
 import static com.ncnf.Utils.SAVED_EVENTS_KEY;
 import static com.ncnf.Utils.USERS_COLLECTION_KEY;
+import static com.ncnf.Utils.UUID_KEY;
+import static com.ncnf.utilities.InputValidator.isStringEmpty;
 
 public class User {
 
     private DatabaseService db;
+
+    private final String uuid;
+    private String username;
     private String email;
-    private final String UUID;
-    private final String path;
-    private Map<String, Object> userData;
+    private String firstName;
+    private String lastName;
+    private List<String> friendsIds;
+    private List<String> ownedEventsIds;
+    private List<String> savedEventsIds;
+    private Date birthDate;
+    private boolean notifications;
+
 
-    
     private final IllegalStateException wrongCredentials = new IllegalStateException("User doesn't have the right credentials to perform current operation");
 
+    public User(DatabaseService db, String uuid, String username, String email, String firstName, String lastName, List<String> friendsIds, List<String> ownedEventsIds, List<String> savedEventsIds, Date birthDate, boolean notifications) {
+        if(isStringEmpty(uuid) || isStringEmpty(email)){
```

Comment:
You should check for all arguements

Reply:
This is a basic Constructor, and some of the fields can be empty in the database. The only fields we know are non empty are those two.